src/commands/projects.ts

```diff
@@ -172,13 +194,33 @@ const deleteProject = async (props: CommonProps & IdOrNameProps) => {
 const update = async (
   props: CommonProps &
     IdOrNameProps & {
-      name: string;
+      name?: string;
+      ipAllowIps?: string[];
+      ipAllowPrimaryBranchOnly?: boolean;
     },
 ) => {
+  const project: ProjectUpdateRequest['project'] = {};
+  if (props.name) {
+    project.name = props.name;
+  }
+  if (props.ipAllowIps) {
+    project.settings = {
+      allowed_ips: {
+        ips: props.ipAllowIps,
+        primary_branch_only: props.ipAllowPrimaryBranchOnly ?? false,
```

Comment:
this would mean that if you have "primary_branch_only" setting set to true it would be overwritten to false, if you don't include this option. I'd say it should remain untouched

Reply:
Also I think current approach would override all settings in the project. Can you validate that?

Reply:
I have to pass `allowed_ips` as a complete object field, I can not change individual `ips` or `primary_branch_only` fields from this config. 

Also, it only overrides `settings.allowed_ips`, not other setting fields. Checked the backend code.

Reply:
Now that I think of if the user only specifies `--ip-allow-primary-branch-only` parameter, it will set `ips` to an empty array.  Resulting in removing all existing configured IPs.  

Should I throw error when only `--ip-allow-primary-branch-only` parameter is specified with no `--ip-allow-ips` 🤔 
(could not find a way in `yargs` for such parameter dependencies)

Reply:
I'd suggest to merge with current settings in the cli code